graylog2-shared/src/main/java/org/graylog2/shared/users/Role.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.shared.users;
+
+import com.google.common.base.Function;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Set;
+
+public interface Role {
+    String getId();
+
+    String getName();
+
+    void setName(String name);
+
+    String getDescription();
+
+    void setDescription(String description);
+
+    Set<String> getPermissions();
+
+    void setPermissions(Set<String> permissions);
+
+    boolean isReadOnly();
+
+    class RoleIdToNameFunction implements Function<String, String> {
```

Comment:
Could we extract these function classes into a helper class (name suggestion `Roles`, similar to `Iterables` or `Sets`) instead of polluting the interface?